pkg/watches/watches_test.go

```diff
@@ -15,195 +15,291 @@
 package watches
 
 import (
+	"bytes"
 	"io/ioutil"
 	"os"
-	"reflect"
-	"testing"
-)
 
-type testCase struct {
-	name            string
-	data            string
-	env             map[string]string
-	expectLen       int
-	expectErr       bool
-	expectOverrides []map[string]string
-}
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
 
-// TODO(joelanford): convert to ginkgo/gomega
-func TestLoadWatches(t *testing.T) {
-	testCases := []testCase{
-		{
-			name: "valid",
-			data: `---
+var _ = Describe("LoadReader", func() {
+	var (
+		expectedWatches   []Watch
+		data              string
+		falseVal, trueVal = false, true
+	)
+	It("should create valid watches", func() {
+		data = `---
 - group: mygroup
   version: v1alpha1
   kind: MyKind
-  chart: ../../testdata/test-chart-0.1.0.tgz
+  chart: ../../testdata/test-chart
   watchDependentResources: false
-  reconcilePeriod: 10s
   overrideValues:
     key: value
-`,
-			expectLen:       1,
-			expectErr:       false,
-			expectOverrides: []map[string]string{{"key": "value"}},
-		},
-		{
-			name: "valid with override expansion",
-			data: `---
+`
+		expectedWatches = []Watch{
+			{
+				GroupVersionKind:        schema.GroupVersionKind{Group: "mygroup", Version: "v1alpha1", Kind: "MyKind"},
+				ChartPath:               "../../testdata/test-chart",
+				WatchDependentResources: &falseVal,
+				OverrideValues:          map[string]string{"key": "value"},
+			},
+		}
```

Comment:
Perhaps update/include some tests with `MaxConcurrentReconciles` and `ReconcilePeriod` being set?